File: backend/api/app/domain/use_cases/message/delete.py
from core.use_cases.use_case import UseCase
from app.domain.repositories.message import BaseRepository
from core.common.equatable import Equatable
from core.common.either import Either
from core.errors.failure import Failure
from app.domain.entities.message import Message

class Params(Equatable):
    def __init__(self, message_id: str) -> None:
        self.message_id = message_id

class DeleteMessage(UseCase[Message]):
    def __init__(self, repository: BaseRepository):
        self.repository = repository
    
    async def __call__(self, params: Params) -> Either[Failure, Message]:
        return await self.repository.delete_message(params.message_id)

File: mobile/lib/features/architect/domains/use_cases/team/delete.dart
import 'package:dartz/dartz.dart';
import 'package:equatable/equatable.dart';

import '../../../../../core/errors/failure.dart';
import '../../../../../core/use_cases/usecase.dart';
import '../../entities/team.dart';
import '../../repositories/team.dart';

class TeamDelete implements UseCase<Team, Params> {
  final TeamRepository repository;

  TeamDelete(this.repository);

  @override
  Future<Either<Failure, Team>> call(Params params) async {
    return await repository.delete(params.teamId);
  }
}

class Params extends Equatable {
  final String teamId;

  const Params({
    required this.teamId,
  });

  @override
  List<Object?> get props => [teamId];
}
